go-concourse/concourse/team.go

```diff
@@ -11,7 +11,7 @@ import (
 
 type Team interface {
 	Name() string
```

Comment:
I think it might be better if we had an accessor that constructs an `atc.Team` object (which can be used directly in order to return the json flag) and places that are using the `Name()` and `Auth()` accessors can pull it off the `atc.Team` object. I did see you mention this in your PR description so I wanna know your thoughts too!

Reply:
I think that could work, especially with changing to something like
```golang
type team struct {
    atc.Team
    connection internal.Connection
    httpAgent  internal.HTTPAgent
}
```

for the private type.

My hesitation is basically around not wanting to make too many far-reaching changes to the structures, which are used everywhere, _even though at the same time_ I do want to do something to straighten out `atc.Team` vs `db.Team` vs `concourse.Team` vs `db.team` vs `concourse.team`. I'll have a go at doing what you suggest, which would make the marshalling part more natural with the rest of the code.

Reply:
Right I definitely understand your concern, I took a quick peek to see how many places would need to change and I think it might only need to be adjusted in a few fly commands but I could be wrong. If it ends up being a more extensive change you can let me know and we can see where to go about from there?

Reply:
I added a [new commit](https://github.com/concourse/concourse/pull/5996/commits/072c7ce9d1b096a51c57d20b573f01b41afb3c1a) with the suggested accessor implemented, but stopped short of removing the existing `team.Name()` and friends. On trying to push that through, it led to some pretty icky-looking changes to go through the additional layer of indirection in order to get the data. (I've also rebased.)